### PR TITLE
Add endpoint for raw telemetry tlog logging

### PR DESCRIPTION
**Pull request by julled**
For drone operations its important to log the mavlink datastreams in different places along the drone connection in case of drone gets lost. Therefor we need tlog logging as some form of blackbox recording.
Based on prior work into this direction in https://github.com/mavlink-router/mavlink-router/pull/269 i updated the changes to be used in current master branch.

- This PR adds a additional endpoint to write the raw mavlink telemetry data to a .tlog file. 
- The functionality can be used from CLI or config file. 
- It shares LogMode and other functionality with flight stack logging.
- It is tested on a raspberry pi4 with a APM in SITL mode connected to it via serial.
- The tlog file is readable by plot.ardupilot.org/

A exemplary console output:
```bash
searchwing@testi:~/mavlink-router $ ./build/src/mavlink-routerd /dev/ttyAMA0:500000 -l /tmp -T a
mavlink-router version v3-4-g237f561
Could not open conf file '/etc/mavlink-router/main.conf' (No such file or directory)
Opened UART [4]CLI: /dev/ttyAMA0
UART [4]CLI: speed = 500000
Opened TCP Server [5] [::]:5760
Logging target system_id=1 on 00000-2023-07-12_02-42-48.tlog
Logging target system_id=1 on 00001-2023-07-12_02-42-49.bin
1 messages to unknown endpoints in the last 5 seconds
```

Note: currently CI is broken , see https://github.com/mavlink-router/mavlink-router/issues/399

### CODE REVIEW

**Comment by booo** [comment]
Looks good to me!

**Comment by jbeyerstedt** [comment]
Please rebase to let the CI run again.

**Review by jbeyerstedt — APPROVED** [review]
LGTM, but maybe @lucasdemarchi has any objections. I remember, that PR #269 tried to do something similar and had started quite a discussion.

File: src/tlog.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#include <unistd.h>
+
+#include <chrono>
+
+#include <common/log.h>
+#include <common/util.h>
+
+bool TLog::_logging_start_timeout()
+{
+    //_target_system_id = -1;
```
Comment (patrickelectric) [inline src/tlog.cpp:34]: Can we move this development code ?

File: src/tlog.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    return true;
+}
+
+bool TLog::start()
+{
+    if (!LogEndpoint::start()) {
+        return false;
+    }
+
+    return true;
```
Comment (patrickelectric) [inline src/tlog.cpp:44]:
```suggestion
    return LogEndpoint::start();
```

**Review by patrickelectric — APPROVED** [review]
Some minor things, but I would be really happy to have it merged.

**Review by lucasdemarchi — APPROVED** [review]

**Pull request merged**